src/lifecycle_point_stamped_incr.cpp

```diff
@@ -0,0 +1,955 @@
+///
+///  `~/x_max (float) [default 1.0]`      - The maximum output value along that axis of movement
+///  `~/y_max (float) [default 1.0]`      - The maximum output value along that axis of movement
+///  `~/z_max (float) [default 1.0]`      - The maximum output value along that axis of movement
+///  `~/alt_x_max (float) [default 0.25]`      - The alternative maximum output value along that axis of movement
+///  `~/alt_x_max (float) [default 0.25]`      - The alternative maximum output value along that axis of movement
+///  `~/alt_x_max (float) [default 0.25]`      - The alternative maximum output value along that axis of movement
+///
+///  `~/x_flip (bool) [default false]`      - Whether the input for this movement should be flipped
+///  `~/y_flip (bool) [default false]`      - Whether the input for this movement should be flipped
+///  `~/z_flip (bool) [default false]`      - Whether the input for this movement should be flipped
+///
+///  `~/boundary_radius (float) [default 0.0]`      - Radius of the spherical space around the zero position that the robot can move in
+///  `~/lin_rate_chg_fac (float) [default 1.0]`     - Factor to the rate of change for the output's values
+///  `~/x_offset (float) [default 0.0]`             - The offset for the message's zero value
+///  `~/y_offset (float) [default 0.0]`             - The offset for the message's zero value
+///  `~/z_offset (float) [default 0.0]`             - The offset for the message's zero value
+///
+///  `~/always_enable (bool) [default false]`      - Whether control input is always enabled (USE WITH CAUTION)
+///
+///  `~/input_device_config_file (std::string) [default "dualshock4_mapping"]`      - Chosen input device config file
+
+#include "rclcpp/rclcpp.hpp"
+#include "sensor_msgs/msg/joy.hpp"
+#include "geometry_msgs/msg/point_stamped.hpp"
+
+#include <string>
+#include <cmath>
+
+#include "yaml-cpp/yaml.h"
+#include <iostream>
+#include <ament_index_cpp/get_package_share_directory.hpp>
+
+#include <chrono>
+#include <functional>
+#include <memory>
+
+// Additional libraries for Lifecycle nodes
+#include <thread>
+#include <utility>
+#include "lifecycle_msgs/msg/transition.hpp"
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp/publisher.hpp"
+#include "rclcpp_lifecycle/lifecycle_node.hpp"
+#include "rclcpp_lifecycle/lifecycle_publisher.hpp"
+#include "rcutils/logging_macros.h"
+#include "std_msgs/msg/string.hpp"
+
+using std::string;
+
+static geometry_msgs::msg::PointStamped command;
+static sensor_msgs::msg::Joy latest_joy_state;
+static bool fresh_joy_state;
+static bool always_enable = false;
+static const float rate_of_change_denom = 1 * std::pow(10, 5); // USER CAN ADJUST, KEEP IT EXTREMELY SMALL
+
+float curr_x_max = 1.0;
+float curr_y_max = 1.0;
+float curr_z_max = 1.0;
+static float x_max;
+static float y_max;
+static float z_max;
+static float alt_x_max;
+static float alt_y_max;
+static float alt_z_max;
+static float boundary_radius;
+float lin_rate_chg_fac;
```

Comment:
static

Reply:
I keep forgetting why static is useful :p but all global variables are now static